### PR TITLE
Middleware support and authenticated cron jobs

### PR DESCRIPTION
**Pull request by ggoodman**
This is a proof-of-concept for adding cron auth support to wt-cli that relies on [webtask-hacks](https://github.com/ggoodman/webtask-hacks).

Ideally, we make a decision whether we want to fully embrace the middleware pipeline approach and release this as a first-class Webtask utility. If we were to do this, what might be a better approach to cron auth (and many other things) would also be to implement the middleware pipeline as a first-class wt-cli concept.

### CODE REVIEW

File: bin/cron/schedule.js [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 
+const Cli = require('structured-cli');
+const Logs = require('../../lib/logs');
+const PrintCronJob = require('../../lib/printCronJob');
+const ValidateCreateArgs = require('../../lib/validateCreateArgs');
+const WebtaskCreator = require('../../lib/webtaskCreator');
+const _ = require('lodash');
 
-var intervals = {
+const CRON_AUTH_MIDDLEWARE = '@webtask/cron-auth-middleware';
```
Comment (tjanczuk) [inline bin/cron/schedule.js]: Wasn't this supposed to live in webtask-userspace?

File: lib/validateCreateArgs.js [hunk 2]
```diff
@@ -4,16 +4,23 @@ const Cli = require('structured-cli');
 const Dotenv = require('dotenv');
 const Fs = require('fs');
 const Path = require('path');
+const Util = require('./util');
 const keyValList2Object = require('./keyValList2Object');
-const _ = require('lodash');
+
+const MIDDLWARE_COMPILER = '@webtask/middleware-compiler';
```
Comment (tjanczuk) [inline lib/validateCreateArgs.js]: Ditto

File: bin/cron/schedule.js (hunk 1 above)
Comment (ggoodman) [reply bin/cron/schedule.js]: This _is_ `webtask-userspace`: https://github.com/auth0/webtask-userspace/blob/master/packages/middleware-compiler/package.json#L2

File: lib/validateCreateArgs.js (hunk 2 above)
Comment (ggoodman) [reply lib/validateCreateArgs.js]: See: https://github.com/auth0/wt-cli/pull/147#discussion_r137070605

**Pull request merged**